lib/plugins/package/lib/zipService.js

```diff
@@ -96,22 +96,24 @@ module.exports = {
 
         return BbPromise.all(normalizedFiles.map(this.getFileContentAndStat.bind(this)))
           .then(contents => {
-            _.sortBy(contents, ['filePath']).forEach(file => {
-              const name = file.filePath.slice(prefix ? `${prefix}${path.sep}`.length : 0);
-              let mode = file.stat.mode;
-              if (
-                normalizedFilesToChmodPlusX &&
-                normalizedFilesToChmodPlusX.includes(name) &&
-                file.stat.mode % 2 === 0
-              ) {
-                mode += 1;
-              }
-              zip.append(file.data, {
-                name,
-                mode,
-                date: new Date(0), // necessary to get the same hash when zipping the same content
+            contents
+              .sort((content1, content2) => content1.filePath - content2.filePath)
```

Comment:
As `filePath` is a string I believe we also need to rely on `localeCompare`

Reply:
Just something I wonder, because I have never used `localeCompare` before. Is it like a safe method to compare strings? What if I have 2 strings that contain no special characters, would it be different than using the `-` operator?

Btw, I'm going to update this.

Reply:
Strings in JS are unicode, and `localeCompare` is native method to handle unicode strings, it handles _any_ JS string

Reply:
Thank you for that information.